### PR TITLE
feat(go-sdk): support OpenTelemetry metrics

### PR DESCRIPTION
**Pull request by ewanharris**
## Description

Backports the telemetry changes back to the generator repository.

Also brings some actions versions changes and fixes and issue where we weren't correctly passing the consistency down for a batch check down to the individual checks.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: config/clients/go/template/docs/opentelemetry.md.mustache [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+| `fga-client.response.model_id` | string | Yes                | Authorization model ID that the FGA server used                                   |
+| `fga-client.user`              | string | No                 | User associated with the action of the request for check and list users           |
+| `http.client.request.duration` | int    | No                 | Duration for the SDK to complete the request, in milliseconds                     |
+| `http.host`                    | string | Yes                | Host identifier of the origin the request was sent to                             |
+| `http.request.method`          | string | Yes                | HTTP method for the request                                                       |
+| `http.request.resend_count`    | int    | Yes                | Number of retries attempted, if any                                               |
+| `http.response.status_code`    | int    | Yes                | Status code of the response (e.g., `200` for success)                             |
+| `http.server.request.duration` | int    | Yes                | Time taken by the FGA server to process and evaluate the request, in milliseconds |
```
Comment (rhamzeh) [inline config/clients/go/template/docs/opentelemetry.md.mustache:45]: This should be false by default, same for `url.scheme`, `url.full` and `http.request.method` 🤔


But those are less important - I'm not sure if the config matches the documentation, but it is crucial we do not enable `http.server.request.duration` as a default attribute b/c it has a very high cardinality

(Side-note: does not need to be in this PR)

File: config/clients/go/template/example/example1/go.mod.mustache [hunk 2]
```diff
@@ -7,4 +7,4 @@ require {{gitHost}}/{{gitUserId}}/{{gitRepoId}} v{{packageVersion}}
 require golang.org/x/sync v0.7.0 // indirect
 
 // To reference local build, uncomment below and run `go mod tidy`
-replace {{gitHost}}/{{gitUserId}}/{{gitRepoId}} v{{packageVersion}} => ../../
\ No newline at end of file
+// replace {{gitHost}}/{{gitUserId}}/{{gitRepoId}} v{{packageVersion}} => ../../
```
Comment (rhamzeh) [inline config/clients/go/template/example/example1/go.mod.mustache:10]: Also for a later PR - I think for all examples, we can make local the default

**Review by rhamzeh — CHANGES_REQUESTED** [review]
All the changes in the `dotnet/template/...` can be dropped, these are build artifacts

File: config/clients/go/template/example/example1/go.mod.mustache (hunk 2 above)
Comment (ewanharris) [reply config/clients/go/template/example/example1/go.mod.mustache:10]: What dya think about eventually getting rid of things like `{{gitHost}}/{{gitUserId}}/{{gitRepoId}}` and things like `{{appshortname}}` in future too now that we have a single SDK publish target? 🥺

File: config/clients/go/template/docs/opentelemetry.md.mustache (hunk 1 above)
Comment (ewanharris) [reply config/clients/go/template/docs/opentelemetry.md.mustache]: Pushed 08996ce202b49254dcf4b9d059d9f1a645a17c61 to update the file with the table/preamble currently used on `main` in go-sdk.

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**